tools/sdk-generation-pipeline/pipelines/steps/parse-and-publish-result.yml

```diff
@@ -82,32 +86,19 @@ steps:
           prNumber=${{ parameters.sdkPR.prNumber }}
           prLink=${{ parameters.sdkPR.prLink }}
         else
-          sdkRepoPullsApi=$(echo $(sdkRepoUrl) | sed "s#https://#https://api.#")
-          sdkRepoPullsApi=$(echo ${sdkRepoPullsApi} | sed "s#$(SDK_OWNER)#repos/$(SDK_PUSH_OWNER)#")
-          sdkRepoPullsApi=$(echo ${sdkRepoPullsApi} | sed "s#.git\$#/pulls#")
-          echo "will use api:${sdkRepoPullsApi}"
-          RESULT=$(curl \
-            -u "$(REPO_OWNER)":"$(azuresdk-github-pat)" \
-            -H "Accept: application/vnd.github.v3+json" \
-            -X POST \
-            -d '{"title":"[AutoPR-${{ parameters.service }}]${{ parameters.swagger.title }}","draft":true,"body":"Create to sync ${{ parameters.swagger.path }}/pull/${{ parameters.swagger.pullRequest }}","base":"'$(sdkMainBranch)'", "head":"$(SDK_PUSH_OWNER):'${branchName}'"}' \
-            ${sdkRepoPullsApi})
-          echo "creat pr finished:$?"
-
-          prNumber=`echo ${RESULT} | jq -r '.number'`
-          prLink=`echo ${RESULT} | jq -r '.html_url'`
-          if [ 10 -gt ${#prLink} ];then
-            sdkRepoPullsList="${sdkRepoPullsApi}?state=open&head=$(SDK_PUSH_OWNER):${branchName}&base=$(sdkMainBranch)"
-            RESULT=$(curl \
-              -u "$(REPO_OWNER)":"$(azuresdk-github-pat)" \
-              -H "Accept: application/vnd.github.v3+json" \
-              -X GET \
-              ${sdkRepoPullsList})
-            if [[ `echo ${RESULT} | jq -r '. | length'` -gt 0 ]]; then
-              prNumber=`echo ${RESULT} | jq -r '.[0].number'`
-              prLink=`echo ${RESULT} | jq -r '.[0].html_url'`
-            fi
-          fi
+          pwsh $(System.DefaultWorkingDirectory)/eng/common/scripts/Submit-PullRequest.ps1 \
+            -RepoOwner "$(SDK_OWNER)" \
+            -RepoName "$(sdkRepoName)" \
+            -BaseBranch "$(sdkMainBranch)" \
+            -PROwner "$(SDK_PUSH_OWNER)" \
+            -PRBranch "${branchName}" \
+            -AuthToken "$(azuresdk-github-pat)" \
+            -PRTitle "[AutoPR-${{ parameters.service }}]${{ parameters.swagger.title }}" \
+            -PRBody "Create to sync ${{ parameters.swagger.path }}/pull/${{ parameters.swagger.pullRequest }}" \
+            -OpenAsDraft $${{ parameters.OpenAsDraft }}
+          
+          prNumber="$(Submitted.PullRequest.Number)"
```

Comment:
As we chatted if you need this information you cannot read it from that variable because it won't be available until after this devop step is complete.

Based on what I understand we might need to add another option to the script that will output the pull request information to a file. Perhaps something like -OutputPullRequestInfo = "file path". Then have the script dump that PR response to that file if it is set.